### PR TITLE
docker: Fix #1303

### LINKED ISSUE #1303: Do not download python package from pip when using RHEL
https://bugzilla.redhat.com/show_bug.cgi?id=1420855

@guits, can you try this one?

### PR DESCRIPTION
**Pull request by guits**
Install package from official repos rather than pip when using RHEL.

This commit fix https://bugzilla.redhat.com/show_bug.cgi?id=1420855

Also this commit Refact all `roles/ceph-*/tasks/docker/pre_requisite.yml`
to avoid a lot of duplicated code.

Fix: #1303
Signed-off-by: Guillaume Abrioux <gabrioux@redhat.com>

### CODE REVIEW

**Comment by guits** [comment]
jenkins test ansible2.2-docker_cluster

**Comment by guits** [comment]
jenkins test ansible2.2-docker_cluster

**Comment by leseb** [comment]
@ktdreyer I think you're going to like this one :)

File: roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml [hunk 1]
```diff
@@ -1,35 +1,76 @@
...
+    with_pkg
+
+- name: install docker on debian
+  apt:
+    name: "{{ item }}"
+    state: present
+    update_cache: yes
+  with_items:
     - docker-engine
+    - python-pip
```
Comment (andrewschoen) [inline roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml]: Why are we installing ``python-pip`` here again?

File: roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml [hunk 2]
```diff
@@ -123,27 +137,30 @@
...
-- name: install docker-py
+- name: install docker-py via pip for every distribution except rhel (ansible < 2.1.0.0)
   pip:
     name: docker-py
     version: 1.1.0
   tags:
     with_pkg
-  when: ansible_version['full'] | version_compare('2.1.0.0', '<')
+  when:
+    - ansible_version['full'] | version_compare('2.1.0.0', '<')
```
Comment (andrewschoen) [inline roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml]: Should we care about ansible versions < 2.2.1 ? We only test ``ansible==2.2.1`` with our CI tests currently.

File: roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml [hunk 3]
```diff
@@ -1,35 +1,76 @@
...
     - python-pip
-    - docker
     - docker.io
   when: ansible_distribution == 'Ubuntu'
   tags:
     with_pkg
 
-- name: install pip and docker on debian
+- name: install pip from testing on debian
   apt:
```
Comment (leseb) [inline roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml]: why don't we use `package` just like in the new tasks?

File: roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml [hunk 4]
```diff
@@ -1,35 +1,76 @@
...
+    default_release: testing
     update_cache: yes
   with_items:
     - python-pip
+  when: ansible_distribution == 'Debian'
+  tags:
+    with_pkg
+
+- name: install docker on debian
+  apt:
```
Comment (leseb) [inline roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml]: why don't we use `package` just like in the new tasks?

File: roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml (hunk 2 above)
Comment (leseb) [inline roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml]: +1

**Comment by ktdreyer** [comment]
There are a several changes in this PR beyond what is described in https://bugzilla.redhat.com/1420855

Why drop the reference to needing docker-py >= 1.1.0 for https://github.com/ansible/ansible-modules-core/issues/1227 ?

Why use docker from apt.dockerproject.org ?

Why use six from PyPI on Ubuntu? Xenial already has v1.10.0,  http://packages.ubuntu.com/xenial/python-six

**Comment by guits** [comment]
@ktdreyer thank you for your feedback.

That's correct, ubuntu Xenial already has 1.10.0 for python-six.
Using docker from apt.dockerproject.org is only for installation on Debian because it's not available in Debian's repositories 

I'm going to update the PR to fix it in agreement with your comments.
Thanks

**Comment by guits** [comment]
@ktdreyer @leseb @andrewschoen, I pushed a new commit, let me know if you have any comment

File: roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml [hunk 5]
```diff
@@ -0,0 +1,80 @@
+---
```
Comment (leseb) [inline roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml]: This file is in the `ceph-docker-common` role already, do we really need to call this file with `_docker.yml`? :) Maybe we can just call it `prerequisites.yml`?

Comment (guits) [inline roles/ceph-docker-common/tasks/pre_requisites/prerequisite_docker.yml]: Done!

**Comment by guits** [comment]
jenkins test ansible2.2-dmcrypt_journal

**Comment by andrewschoen** [comment]
Thanks for the refactor of the all the ``pre_requisite.yml`` files from each individual role into ``ceph-docker-common``, it's much nicer that way.

**Comment by leseb** [comment]
I like this: `+161 −1,047`, thanks @guits

**Pull request merged**